rmf_fleet_adapter/src/rmf_fleet_adapter/agv/ReservationManager.hpp

```diff
@@ -24,33 +24,26 @@
 namespace rmf_fleet_adapter {
 namespace agv {
 
+class RobotContext;
 
 class ReservationManager
 {
 public:
-  void add_ticket(const rmf_reservation_msgs::msg::ReservationAllocation alloc)
-  {
-    allocations.push_front(alloc);
-  }
+  /// Adds a ticket and releases the previous ticket if the ticket id is different
+  void add_ticket(
```

Comment:
Very minor nitpick: `add` gives the impression that this will be combined with the previous ticket. What if we call this `replace_ticket` instead?